packages/components/collapse/src/use-collapse.ts

```diff
@@ -0,0 +1,71 @@
+import { computed, provide, ref, watch } from 'vue'
+import { ensureArray } from '@element-plus/utils'
+import { useNamespace } from '@element-plus/hooks'
+import { CHANGE_EVENT, UPDATE_MODEL_EVENT } from '@element-plus/constants'
+import { collapseContextKey } from '@element-plus/tokens'
+
+import type { SetupContext } from 'vue'
+import type {
+  CollapseActiveName,
+  CollapseEmits,
+  CollapseProps,
+} from './collapse'
+
+export const useCollapse = (
+  props: CollapseProps,
+  emit: SetupContext<CollapseEmits>['emit']
+) => {
+  const activeNames = ref(ensureArray(props.modelValue))
+
+  const setActiveNames = (_activeNames: CollapseActiveName[]) => {
+    activeNames.value = _activeNames
+    const value = props.accordion ? activeNames.value[0] : activeNames.value
+    emit(UPDATE_MODEL_EVENT, value)
+    emit(CHANGE_EVENT, value)
+  }
+
+  const handleItemClick = (name: CollapseActiveName) => {
+    if (props.accordion) {
+      setActiveNames([
+        (activeNames.value[0] || activeNames.value[0] === 0) &&
+        activeNames.value[0] === name
+          ? ''
+          : name,
```

Comment:
It seems that it can be simplified as:
```ts
  activeNames.value[0] === name ? '' : name
```

Reply:
Right, good catch.

Reply:
@btea updated, would you mind looking again? Thanks in advance!